packages/cyberstorm-forms/src/components/FormSelect.tsx

```diff
@@ -0,0 +1,126 @@
+"use client";
+
+import { z, ZodObject } from "zod";
+import { ZodRawShape } from "zod/lib/types";
+import { Path, useController } from "react-hook-form";
+import styles from "./FormSelect.module.css";
+import * as Select from "@radix-ui/react-select";
+import React, { PropsWithChildren } from "react";
+import { Icon } from "@thunderstore/cyberstorm";
+import { faCaretDown, faCheckCircle } from "@fortawesome/pro-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+export type FormSelectProps<
+  Schema extends ZodObject<Z>,
+  Z extends ZodRawShape
+> = {
+  // The schema is required to allow TS to infer valid values for the name field
+  schema: Schema;
+  name: Path<z.infer<Schema>>;
+  options: { value: string; label: string }[];
+  placeholder?: string;
+  defaultValue?: string;
+  ariaLabel?: string;
+};
+export function FormSelect<Schema extends ZodObject<Z>, Z extends ZodRawShape>({
+  name,
+  defaultValue,
+  options,
+  ariaLabel,
+}: FormSelectProps<Schema, Z>) {
+  const {
+    field,
+    fieldState: { error },
+    formState: { isSubmitting, disabled },
+  } = useController({ name });
+
+  if (defaultValue) {
+    React.useEffect(() => {
+      field.onChange(defaultValue);
+    }, [defaultValue]);
+  }
+
+  return (
+    <>
+      <TempSelect
```

Comment:
Nitpick: not fan of the name since there's nothing temporary about this.

Main point: is the wrapper even needed? I appreciate the idea of breaking down a large component to subcomponents, but in this particular case I feel it goes too far, and adding the TempSelect-in-the-middle component just makes things unnecessarily complicated.